Source/WebKit/UIProcess/API/Swift/WKUIDelegateAdapter.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2024 Apple Inc. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+// 1. Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+// BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+
+#if ENABLE_SWIFTUI && compiler(>=6.0)
+
+import Foundation
+internal import WebKit_Private
+
+fileprivate struct DefaultDialogPresenting: DialogPresenting {
+}
+
+@MainActor
+final class WKUIDelegateAdapter: NSObject, WKUIDelegate {
+    init(dialogPresenter: (any DialogPresenting)?) {
+        self.dialogPresenter = dialogPresenter ?? DefaultDialogPresenting()
+    }
+
+    private let dialogPresenter: any DialogPresenting
+
+    // MARK: Dialog presentation
+
+    func webView(_ webView: WKWebView, runJavaScriptAlertPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo) async {
+        await dialogPresenter.handleJavaScriptAlert(message: message, initiatedBy: .init(frame))
+    }
+
+    func webView(_ webView: WKWebView, runJavaScriptConfirmPanelWithMessage message: String, initiatedByFrame frame: WKFrameInfo) async -> Bool {
+        let result = await dialogPresenter.handleJavaScriptConfirm(message: message, initiatedBy: .init(frame))
+        return switch result {
+        case .ok:
```

Comment:
Since you are using implicit returns already, you could also shorten this to one line. That's kinda been my logic for these:
- Multiple lines, more complex, write `return`
- Single line, simpler expression, elide it

¯\\\_(ツ)\_/¯

Reply:
so it would be like

```
switch await dialogPresenter... { 
```

?

Reply:
Sorry, I was implying:
```swift
let result = await dialogPresenter.handleJavaScriptConfirm(message: message, initiatedBy: .init(frame))
return switch result {
case .ok: true
case .cancel: false
}
```